maps/GoldenrodUndergroundWarehouse.asm

```diff
@@ -71,8 +71,8 @@ GoldenrodUndergroundWarehouseDirectorScript:
 	opentext
 	checkevent EVENT_RECEIVED_CARD_KEY
 	iftrue .GotCardKey
-	readvar VAR_BADGES
-	ifgreater 6, .GotCardKey
+	checkcode VAR_BADGES
+	ifless 7, .GotCardKey
```

Comment:
this seems suspicious. what happens if the player gets 8 badges and then does the radio tower?

Reply:
Then they receive the "card key" like they normally would. The comparison is used to check for when the player should NOT be receiving the card key (if its true, then it jumps to "GotCardKey" and skips the code that gives the card key).